homeassistant/components/unifi/sensor.py

```diff
@@ -388,6 +407,16 @@ class UnifiSensorEntity(UnifiEntity[HandlerT, ApiItemT], SensorEntity):
 
     entity_description: UnifiSensorEntityDescription[HandlerT, ApiItemT]
 
+    @callback
+    def _make_disconnected(self, *_: core_Event) -> None:
+        """No heart beat by device.
+
+        Reset sensor value to 0 when client device is disconnected
+        """
+        if self._attr_native_value != 0:
+            self._attr_native_value = 0
```

Comment:
I think this falls under calculating state which we don't allow for integrations that integrate devices apr services.

The library should handle this of needed.

Reply:
This is the same logic as how the device tracker calculates availability so there is nothing new introduced with this PR

Reply:
I guess the alternative is to mark the entity unavailable.

Reply:
If the entity represents a feature of a device and that device isn't connected we normally mark the entity as unavailable, yes.

Reply:
Hello, let me know if I should create a new PR.

Reply:
If marking the entity as unavailable on device disconnection sounds good, please make a PR for that.